app/main.py

```diff
@@ -1,3 +1,18 @@
-def cache(func):
-    # Write your code here
-    pass
+import time
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    cache_result = {}
+
+    def add_result(*args, **kwargs) -> Callable:
```

Comment:
It is a common convention to name inner function in decorator as `inner`. Consider to rename it